packages/perseus/src/util/parse-perseus-json/perseus-parsers/widgets-map.test.ts

```diff
@@ -34,6 +34,29 @@ describe("parseWidgetsMap", () => {
         expect(result).toEqual(anyFailure);
     });
 
+    it("rejects a key with ID 0", () => {
```

Comment:
We usually refer to this value as the widget Id and not the "key". I think it's a useful differentiation because it avoids being conflated with a React component key.

Reply:
What are the parts of a widget ID called, then? If the widget ID is something like `radio 1`, what is the `radio` part called and what is the `1` part called?

Reply:
I think I'm just going to call it the "widget number".

Reply:
`radio` is the widget type (in some places its referred to as the widget name). 

`radio 1` is the widget id and I've been working hard to avoid folks thinking of them as being "made up" of a type and number, but rather that we push for them to be opaque identifiers.

Reply:
Hmm, well, they're definitely not treated as opaque currently, because if the number is 0 the entire page crashes. I've tried to avoid referring directly to the number in the tests and variable names.

Reply:
Agreed, we do parse these widget ids in places, but I feel it's a thing we should avoid and remove when we can. Interpreting IDs as informative structures has bit me many times in the past when needs cause us to change the ID in some way. We hit this during Goliath when we assumed that KAIDs were of a specific format (legacy IDs weren't of that format).